File: leetcode/121. Best Time to Buy and Sell Stock/main.java
// Solved by @kitanoyoru
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/submissions/

class Solution {
    public int maxProfit(int[] prices) {
       int buy = Integer.MAX_VALUE, sell = 0;

       for (int price : prices) {
         buy = Math.min(buy, price);
         sell = Math.max(sell, price - buy);
       }

       return sell;
    }
}

File: leetcode/121. Best Time to Buy and Sell Stock/main.py
# Solved by @kitanoyoru
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock/

from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        [buy, sell] = [float("inf"), 0]

        for price in prices:
            buy = min(buy, price)
            sell = max(sell, price - buy)

        return sell
